tutorials/micro/micro_autotune.py

```diff
@@ -0,0 +1,268 @@
+    ############################
+    # For comparison, let's compile and run the graph without imposing any autotuning schedules. TVM
+    # will select a randomly-tuned implementation for each operator, which should not perform as well as
+    # the tuned operator.
+
+    with pass_context:
+        lowered = tvm.relay.build(relay_mod, target=target, params=params)
+
+    temp_dir = tvm.contrib.utils.tempdir()
+    if args.platform == "host":
+        project = tvm.micro.generate_project(
+            str(repo_root / "src" / "runtime" / "crt" / "host"), lowered, temp_dir / "project"
+        )
+
+    else:
+        project = tvm.micro.generate_project(
+            str(repo_root / "apps" / "microtvm" / "zephyr" / "template_project"),
+            lowered,
+            temp_dir / "project",
+            {
+                "zephyr_board": board,
+                "west_cmd": "west",
+                "verbose": 1,
+                "project_type": "host_driven",
+            },
+        )
+
+    project.build()
+    project.flash()
+    with tvm.micro.Session(project.transport()) as session:
+        debug_module = tvm.micro.create_local_debug_executor(
+            lowered.get_graph_json(), session.get_system_lib(), session.device
+        )
+        debug_module.set_input(**lowered.get_params())
+        print("########## Build without Autotuning ##########")
+        debug_module.run()
+        del debug_module
+
+    ##########################
+    # Timing the tuned program
+    ##########################
+    # Once autotuning completes, you can time execution of the entire program using the Debug Runtime:
+
+    with tvm.autotvm.apply_history_best("microtvm_autotune.log"):
+        with pass_context:
+            lowered_tuned = tvm.relay.build(relay_mod, target=target, params=params)
+
+    temp_dir = tvm.contrib.utils.tempdir()
+    if args.platform == "host":
+        project = tvm.micro.generate_project(
+            str(repo_root / "src" / "runtime" / "crt" / "host"), lowered_tuned, temp_dir / "project"
+        )
+
+    else:
+        project = tvm.micro.generate_project(
+            str(repo_root / "apps" / "microtvm" / "zephyr" / "template_project"),
+            lowered_tuned,
+            temp_dir / "project",
+            {
+                "zephyr_board": board,
+                "west_cmd": "west",
+                "verbose": 1,
+                "project_type": "host_driven",
+            },
+        )
+
+    project.build()
+    project.flash()
+    with tvm.micro.Session(project.transport()) as session:
+        debug_module = tvm.micro.create_local_debug_executor(
+            lowered_tuned.get_graph_json(), session.get_system_lib(), session.device
+        )
+        debug_module.set_input(**lowered_tuned.get_params())
+        print("########## Build with Autotuning ##########")
+        debug_module.run()
+        del debug_module
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
```

Comment:
nit: good to have a simple description of the program

Reply:
great point. added!